appinventor/appengine/src/com/google/appinventor/client/explorer/youngandroid/ProjectList.java

```diff
@@ -91,7 +89,7 @@ public ProjectList() {
     sortOrder = SortOrder.DESCENDING;
 
     // Initialize UI
-    table = new Grid(1, 5); // The table initially contains just the header row.
+    table = new Grid(3, 5); // The table initially contains just the header row.
```

Comment:
Any reason as to why this was changed?

Reply:
The Grid object only seems to have an insertRow(before) method, not an append method. insertRow requires an additional row at the end so that there is a valid row for rows to be inserted before.

The rows are then:
0: Header row
1: First row that will receive data when the grid is refreshed
2: Extra row so that I can insert rows before it

It's possible I'm somehow violating the intended paradigm of the Grid object because this seemed pretty awkward. If there's an append someplace I missed, please point me to it.

Reply:
Like I said, it works, but it just wasn't clear to me why it worked, especially given that the original code started with a single row.

Reply:
refreshTable sets the Grid to a minimum of 3 rows, so it's not strictly necessary to do it in the initialization, but that's the smallest the Grid can be.

Probably worth mentioning this in a comment, since it's not intuitive.